### PR TITLE
Dynamic font sizes part 2

### PR DESCRIPTION
**Pull request by NicolasPetton**
Set paddings and font sizes based on `vh` in presentation mode only

* In "normal" mode, use `rem` units for font sizes
* In presentation mode, use `vh` units to scale fonts
* Remove theme font-size/line-height definitions set in px (relative font-size/line-height definitions are kept)

With the new scaled font-size, setting font sizes in px does not work anymore.

@kookma Could you checkout this branch and try it? I've played with it both on mobile and desktop, and I like the result so far, but more tweaking and testing might be needed.

With these changes, the number of lines per slide is **almost** constant in presentation and normal mode. Getting them to scale exactly the same way is tricky, as paddings/line heights and other factors come into play, but I think this is pretty close.

I'm not 100% convinced with the left/right padding on presentation mode, especially on mobile, so I might try to improve that part in a later PR.

### CODE REVIEW

**Comment by kookma** [comment]
Thank you  @NicolasPetton - I will make some experiments and ;et you know the result.

**Comment by kookma** [comment]
@NicolasPetton I did some experiments with the current presentation example in Tamasha. The demo/Intro presentation has a Long Long Tid slide which has longer text than other slide so I may act as the worse case.

1. On deskktop (1080*1920) there is no overflow, but text seems a little small.
2. For all screen size where width is greater than height (including the rotated mobile screen where height > width) there is no overflow
3. For all screen with height>width there is overflow

I may suggest using `clamp` to limit font size stay in a acceptable range.

**Comment by kookma** [comment]
In your post above for Desktop (1280px wide) presentation the text has flowed over footnote. Why padding does not work here?

**Comment by NicolasPetton** [comment]
> I may suggest using clamp to limit font size stay in a acceptable range.

I'm fine with that, however we'll lose the linear scaling.

> In your post above for Desktop (1280px wide) presentation the text has flowed over footnote. Why padding does not work here?

I'll have a look.

**Comment by kookma** [comment]
> I'm fine with that, however we'll lose the linear scaling.

I thought using `clamp (lower, linear-function, upper) ` we can keep the linear scaling? Am I wrong?

**Comment by kookma** [comment]
@NicolasPetton - please see https://css-tricks.com/linearly-scale-font-size-with-css-clamp-based-on-the-viewport/
for linear scaling font using clamp. What do you think?

**Comment by NicolasPetton** [comment]
> I thought using clamp (lower, linear-function, upper) we can keep the linear scaling? Am I wrong?

Yes and no, linear scaling will work up until you reach the clamp thresholds.

I'll play more with this tonight and will have a look at the paddings as well.

**Comment by NicolasPetton** [comment]
@kookma if you think the font size is too small in general, we could make it bigger (proportionally with the viewport height). The question is, how many lines do we want to have per slide without overflow?

**Comment by kookma** [comment]
> @kookma if you think the font size is too small in general, we could make it bigger (proportionally with the viewport height). The question is, how many lines do we want to have per slide without overflow?

I have pushed 0.4.3 update. I tested on a full HD monitor and I feel the font are a bit small but the good news is there is no overflow. As I said before the Long Long Tid is a worst case slide and in real presentation we do not have such a slide. see best practice for presentation https://kookma.github.io/TW-Tamasha/#demo%2FBest%20Practice.

I will announce on the group and ask for feedback, specially I like people test using projectors on big screen.

To sum up for now:

1. I prefer to use `clamp` to limit scaling between two ends
2. I think we need to use a little bigger fonts but not too big
3. We may use font weight to compensate the small size for better readability

side note: I have add some classes to hide UI element. I also changed 600px height to 480px in normal view! Please let me know your comments here.

**Comment by NicolasPetton** [comment]
> I also changed 600px height to 480px in normal view

This means that the number of visible lines in a slide in normal mode will be quite different from presentation mode. Ideally I'd like them to be at least similar, so that white working on the slides authors have an idea of the final result when presenting slides.

I'll make the font size a bit bigger in presentation mode, hopefully that'll help with the reduced size in normal mode.

**Comment by kookma** [comment]
> This means that the number of visible lines in a slide in normal mode will be quite different from presentation mode. Ideally I'd like them to be at least similar, so that white working on the slides authors have an idea of the final result when presenting slides.

Agree!  That was for the case to embed a presentation inside a tiddler with other contents! I even thought to be able to overwrite it for normal mode! But having the same number of lines in both mode is more important!

**Comment by kookma** [comment]
I think 5 to maximum 7 lines is enough per slide! Five points (list item) is preferred!
